default/python/AI/FleetUtilsAI.py

```diff
@@ -68,107 +68,97 @@ def get_targeted_planet_ids(planet_ids, mission_type):
+    :rtype: list[int]
     """
-    if verbose:
-        print ("get_fleets_for_mission: (nships:%1d, targetStats:%s, minStats:%s, "
-               "curStats:%s, species:%6s, systemsToCheck:%8s, systemsChecked:%8s, "
-               "fleetPoolSet:%8s, fleetList:%8s) ") % (nships, target_stats, min_stats, cur_stats, species,
-                                                       systems_to_check, systems_checked, fleet_pool_set, fleet_list)
     universe = fo.getUniverse()
-    if not (systems_to_check and fleet_pool_set):
-        if verbose:
-            print "no more systems or fleets to check"
-        if take_any:
-            return fleet_list
-        elif stats_meet_reqs(cur_stats, min_stats) and sum(len(universe.getFleet(fid).shipIDs) for fid in fleet_list) >= nships:
-            return fleet_list
-        else:
-            return []
-    this_system_id = systems_to_check.pop(0)  # take the head of the line
-    systems_checked.append(this_system_id)
-
-    accessible_fleets = foAI.foAIstate.systemStatus.get(this_system_id, {}).get('myFleetsAccessible', [])
-    fleets_here = [fid for fid in accessible_fleets if fid in fleet_pool_set]
-
-    if verbose:
-        print "found fleetPool Fleets %s" % fleets_here
-    while fleets_here:
-        fleet_id = fleets_here.pop(0)
-        fleet = universe.getFleet(fleet_id)
-        if not fleet:
-            print "in get_fleets_for_mission, fleet_id %d appears invalid; cannot retrieve" % fleet_id
-            fleet_pool_set.remove(fleet_id)
-            continue
-        if len(list(fleet.shipIDs)) > 1:
-            new_fleets = split_fleet(fleet_id)  # try splitting fleet
-            fleet_pool_set.update(new_fleets)
-            fleets_here.extend(new_fleets)
-        meets_species_req = False
-        needs_species = False
-        if species != "":
-            needs_species = True
-        has_species = ""
-
-        colonization_roles = (ShipRoleType.CIVILIAN_COLONISATION,
-                              ShipRoleType.BASE_COLONISATION)
-        for shipID in fleet.shipIDs:
-            ship = universe.getShip(shipID)
-            if foAI.foAIstate.get_ship_role(ship.design.id) in colonization_roles:
-                has_species = ship.speciesName
-                if has_species == species:
-                    meets_species_req = True
-                    break
-        troop_capacity = 0
-        needs_troops = 'troopCapacity' in target_stats
-        if needs_troops:
-            troop_capacity = count_troops_in_fleet(fleet_id)
-
-        use_fleet = (((not needs_species and not has_species) or meets_species_req) and
-                     ((needs_troops and troop_capacity > 0) or (not needs_troops and not troop_capacity)))
-        if use_fleet:
+    colonization_roles = (ShipRoleType.CIVILIAN_COLONISATION, ShipRoleType.BASE_COLONISATION)
+    systems_to_check = [starting_system]
+    systems_checked = []
```

Comment:
I simply used the previous variable name without much thought. ```visited``` seems fine to me.